unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/TextShape/ECSTextShapeComponent.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using DCL.ECSRuntime;
+using Google.Protobuf;
+using UnityEngine;
+
+namespace DCL.ECSComponents
+{
+    public class ECSTextShapeComponent : IDisposable
+    {
+        private readonly ECSComponentsFactory factory;
+        private readonly IECSComponentWriter componentWriter;
+        private readonly int componentId;
+
+        public ECSTextShapeComponent(int componentId, ECSComponentsFactory factory, IECSComponentWriter componentWriter)
+        {
+            ECSTextShapeComponentHandler handler = new ECSTextShapeComponentHandler(DataStore.i.ecs7, AssetPromiseKeeper_Font.i);
+            factory.AddOrReplaceComponent(componentId, TextShapeSerialization.Deserialize, () => handler);
```

Comment:
This will use a single handler instance for all TextShape component, but I see that the handler has a state:
```
    internal GameObject textGameObject;
    internal TextMeshPro textComponent;
    internal RectTransform rectTransform;
    internal AssetPromise_Font promise;
    private PBTextShape currentModel;
```
that means that every TextShape will share the same `textGameObject`, for example, and that it will be overwritten with a new reference every time a component is created
```
    public void OnComponentCreated(IParcelScene scene, IDCLEntity entity)
    {
        textGameObject = new GameObject(COMPONENT_NAME);
        ...
    }
```